sotodlib/coords/helpers.py

```diff
@@ -228,11 +243,33 @@ def get_wcs_kernel(proj, ra, dec, res):
     Returns a WCS object that captures the requested pixelization.
 
     """
-    assert np.isscalar(res)  # This ain't enlib.
-    _, wcs = enmap.geometry(np.array((dec, ra)), shape=(1,1), proj=proj,
-                            res=(res, -res))
+    if len(proj) > 3:
+        m = re.match('(?P<proj>car|tan)-(?P<res>[0-9]*.?[0-9]*)(?P<unit>arcsec|arcmin|deg)', proj)
```

Comment:
If the expression doesn't match, then m will be None.  Check for that, and raise a ValueError saying the string didn't parse.  Otherwise the next line will error and the user will just be confused.